opslevel/resource_opslevel_trigger_definition.go

```diff
@@ -268,12 +285,17 @@ func (r *TriggerDefinitionResource) Update(ctx context.Context, req resource.Upd
 		return
 	}
 
-	extendedTeams, err := updatedTriggerDefinition.ExtendedTeamAccess(r.client, nil)
-	if err != nil {
-		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("Unable to get teams for 'extended_team_access', got error: %s", err))
-		return
+	if planModel.ExtendedTeamAccess.IsNull() || planModel.ExtendedTeamAccess.IsUnknown() {
+		extendedTeamsAccess = types.ListNull(types.StringType)
+	} else if len(planModel.ExtendedTeamAccess.Elements()) == 0 {
+		extendedTeamsAccess = types.ListValueMust(types.StringType, []attr.Value{})
+	} else {
+		extendedTeamsAccess, err = getExtendedTeamAccessListValue(r.client, updatedTriggerDefinition)
+		if err != nil {
+			resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("Unable to get teams for 'extended_team_access', got error: %s", err))
+			return
+		}
```

Comment:
maybe i'm missing something subtle but this seems duplicated 3 times - what was the reason for not putting this into `NewTriggerDefinitionResourceModel` ?

Reply:
Originally the function signature for `NewTriggerDefinitionResourceModel()` was getting really long, but it's not that bad now. I'll be consolidating these 3 into 1